php/src/Google/Protobuf/Internal/GPBUtil.php

```diff
@@ -70,15 +75,13 @@ public function divideInt64ToInt32($value, &$high, &$low, $trim = false)
     public static function checkString(&$var, $check_utf8)
     {
         if (is_array($var) || is_object($var)) {
-            trigger_error("Expect string.", E_USER_ERROR);
-            return;
+            throw new \Exception("Expect string.");
```

Comment:
Prefer `InvalidArgumentException`, since this function may also be used in other situations besides decode.